imap_processing/cdf/utils.py

```diff
@@ -159,7 +164,6 @@ def write_cdf(dataset: xr.Dataset):
     xarray_to_cdf(
         dataset,
         str(file_path),
-        datetime64_to_cdftt2000=True,
```

Comment:
Does the `int64` epoch get written correctly in the CDF as datatype: CDF_TIME_TT2000?

Reply:
Apparently so... I just dumped our test dataset out to disk and ran `cdfdump` on it. One concerning thing is that our MET times in that test file are 1, 2, 3. But we are getting
```
  Record # 1: 2009-12-31T23:59:59.000000000
  Record # 2: 2010-01-01T00:00:00.000000000
  Record # 3: 2010-01-01T00:00:01.000000000
```
But our IMAP_EPOCH is set to 2010-01-01T00:00, so those seem off to me. Are we missing a nanoseconds conversion somewhere / am I missing something obvious?

```
Variable Information (0 rVariable, 1 zVariable)
===========================================================
epoch                 CDF_TT2000/1      0:[]    T/  


Variable (1 variable)
=========================================

epoch (No: 0) (Recs: 3)
-----
Data Type:           CDF_TT2000
Dimensionality:      0:[]       (T/)  
Format:              
Pad value:           0000-01-01T00:00:00.000000000
Written Records:     3/3(max)
Allocated Records:   3/3(max)
Blocking Factor:     1 (records)
Attribute Entries:
     CATDESC         (CDF_CHAR/66): "Time, number of nanoseconds since J2000 with leap seconds included"
     FIELDNAM        (CDF_CHAR/5): "epoch"
     FILLVAL         (CDF_TT2000/1): 9999-12-31T23:59:59.999999999
     LABLAXIS        (CDF_CHAR/5): "epoch"
     FORMAT          (CDF_CHAR/1): ""
     UNITS           (CDF_CHAR/2): "ns"
     VALIDMIN        (CDF_TT2000/1): 1990-01-01T00:00:00.000000000
     VALIDMAX        (CDF_TT2000/1): 2099-12-31T00:00:00.000000000
     VAR_TYPE        (CDF_CHAR/12): "support_data"
     SCALETYP        (CDF_CHAR/6): "linear"
     MONOTON         (CDF_CHAR/8): "INCREASE"
     TIME_BASE       (CDF_CHAR/5): "J2000"
     TIME_SCALE      (CDF_CHAR/16): "Terrestrial Time"
     REFERENCE_POSITION (CDF_CHAR/20): "Rotating Earth Geoid"
     DEPEND_0        (CDF_CHAR/5): "epoch"
Variable Data:
  Record # 1: 2009-12-31T23:59:59.000000000
  Record # 2: 2010-01-01T00:00:00.000000000
  Record # 3: 2010-01-01T00:00:01.000000000
```

Reply:
I believe that you are on the right track asking about leapseconds. My reading of the numpy documentation is that conversion to/from a string is for TAI time, not UTC. See: https://numpy.org/doc/stable/reference/arrays.datetime.html#datetime64-conventions-and-assumptions

Comparing to SPICE, this is what I get:
```
from emmemuspy.utils.time import utc_2_et
IMAP_EPOCH = np.datetime64("2010-01-01T00:00:00", "ns")
J2000_EPOCH = np.datetime64("2000-01-01T11:58:55.816", "ns")
>>> (IMAP_EPOCH - J2000_EPOCH).astype("timedelta64[ns]").astype(np.int64) / 1e9 - utc_2_et("2010-01-01T00:00:00")
-1.9999244809150696
```
This is consistent with the two leapseconds that occured between 2000 and 2010 (one in 2005 and one in 2008).